### ISSUE TITLE
Dask 0.20.0 triggers failure with atop 

### ISSUE THREAD

**Issue by jakirkham** [nanshe-org/nanshe_workflow#269]
When trying to determine the registration shifts, it appears that Dask 0.20.0 fails when computing [these lines]( https://github.com/nanshe-org/nanshe_workflow/blame/9ee2f1529ab7f6b3a21a8483a748c01c37ebe785/nanshe_ipython.ipynb#L650-L657 ). This can be verified by trying to call `compute` on `da_shifts`. Here's the traceback. More details in [this log]( https://app.wercker.com/nanshe-org/nanshe_workflow/runs/build/5bdd31ee1645fe0007cf9e0a?step=5bdd3257591ca8000776de0f ).

<details>

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-c46338dd2b6d> in <module>
     91         new_axes={da_overlap_max_match.ndim: da_overlap_max_match.ndim - 1}
     92     )
---> 93     da_shifts.compute()
     94 
     95     # Free connected persisted values

/opt/conda3/lib/python3.6/site-packages/dask/base.py in compute(self, **kwargs)
    153         dask.base.compute
    154         """
--> 155         (result,) = compute(self, traverse=False, **kwargs)
    156         return result
    157 

/opt/conda3/lib/python3.6/site-packages/dask/base.py in compute(*args, **kwargs)
    390     keys = [x.__dask_keys__() for x in collections]
    391     postcomputes = [x.__dask_postcompute__() for x in collections]
--> 392     results = schedule(dsk, keys, **kwargs)
    393     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    394 

/opt/conda3/lib/python3.6/site-packages/distributed/client.py in get(self, dsk, keys, restrictions, loose_restrictions, resources, sync, asynchronous, direct, retries, priority, fifo_timeout, **kwargs)
   2306             try:
   2307                 results = self.gather(packed, asynchronous=asynchronous,
-> 2308                                       direct=direct)
   2309             finally:
   2310                 for f in futures.values():

/opt/conda3/lib/python3.6/site-packages/distributed/client.py in gather(self, futures, errors, maxsize, direct, asynchronous)
   1645             return self.sync(self._gather, futures, errors=errors,
   1646                              direct=direct, local_worker=local_worker,
-> 1647                              asynchronous=asynchronous)
   1648 
   1649     @gen.coroutine

/opt/conda3/lib/python3.6/site-packages/distributed/client.py in sync(self, func, *args, **kwargs)
    663             return future
    664         else:
--> 665             return sync(self.loop, func, *args, **kwargs)
    666 
    667     def __repr__(self):

/opt/conda3/lib/python3.6/site-packages/distributed/utils.py in sync(loop, func, *args, **kwargs)
    275             e.wait(10)
    276     if error[0]:
--> 277         six.reraise(*error[0])
    278     else:
    279         return result[0]

/opt/conda3/lib/python3.6/site-packages/six.py in reraise(tp, value, tb)
    691             if value.__traceback__ is not tb:
    692                 raise value.with_traceback(tb)
--> 693             raise value
    694         finally:
    695             value = None

/opt/conda3/lib/python3.6/site-packages/distributed/utils.py in f()
    260             if timeout is not None:
    261                 future = gen.with_timeout(timedelta(seconds=timeout), future)
--> 262             result[0] = yield future
    263         except Exception as exc:
    264             error[0] = sys.exc_info()

/opt/conda3/lib/python3.6/site-packages/tornado/gen.py in run(self)
   1097 
   1098                     try:
-> 1099                         value = future.result()
   1100                     except Exception:
   1101                         self.had_exception = True

/opt/conda3/lib/python3.6/site-packages/tornado/gen.py in run(self)
   1105                     if exc_info is not None:
   1106                         try:
-> 1107                             yielded = self.gen.throw(*exc_info)
   1108                         finally:
   1109                             # Break up a reference to itself

/opt/conda3/lib/python3.6/site-packages/distributed/client.py in _gather(self, futures, errors, direct, local_worker)
   1490                             six.reraise(type(exception),
   1491                                         exception,
-> 1492                                         traceback)
   1493                     if errors == 'skip':
   1494                         bad_keys.add(key)

/opt/conda3/lib/python3.6/site-packages/six.py in reraise(tp, value, tb)
    690                 value = tp()
    691             if value.__traceback__ is not tb:
--> 692                 raise value.with_traceback(tb)
    693             raise value
    694         finally:

/opt/conda3/lib/python3.6/site-packages/dask/optimization.py in __call__()
    940                              % (len(self.inkeys), len(args)))
    941         return _get_recursive(self.dsk, self.outkey,
--> 942                               dict(zip(self.inkeys, args)))
    943 
    944     def __reduce__(self):

/opt/conda3/lib/python3.6/site-packages/dask/core.py in _get_recursive()
    130         return cache[x]
    131     elif hashable and x in dsk:
--> 132         res = cache[x] = _get_recursive(dsk, dsk[x], cache)
    133         return res
    134     elif type(x) is tuple and x and callable(x[0]):  # istask

/opt/conda3/lib/python3.6/site-packages/dask/core.py in _get_recursive()
    134     elif type(x) is tuple and x and callable(x[0]):  # istask
    135         func, args = x[0], x[1:]
--> 136         args2 = [_get_recursive(dsk, k, cache) for k in args]
    137         return func(*args2)
    138     return x

/opt/conda3/lib/python3.6/site-packages/dask/core.py in <listcomp>()
    134     elif type(x) is tuple and x and callable(x[0]):  # istask
    135         func, args = x[0], x[1:]
--> 136         args2 = [_get_recursive(dsk, k, cache) for k in args]
    137         return func(*args2)
    138     return x

/opt/conda3/lib/python3.6/site-packages/dask/core.py in _get_recursive()
    130         return cache[x]
    131     elif hashable and x in dsk:
--> 132         res = cache[x] = _get_recursive(dsk, dsk[x], cache)
    133         return res
    134     elif type(x) is tuple and x and callable(x[0]):  # istask

/opt/conda3/lib/python3.6/site-packages/dask/core.py in _get_recursive()
    135         func, args = x[0], x[1:]
    136         args2 = [_get_recursive(dsk, k, cache) for k in args]
--> 137         return func(*args2)
    138     return x
    139 

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

</details>

<br>

Interestingly calling `compute` on `da_overlap_max_match` preceding works without issues. This in spite of the fact that the error seems to indicate treating an array as a `bool` results in an issue.

The error does not show up in Dask 0.19.4 or any version previous to that. Also this code in the workflow has not changed in a very long time. So this is probably some kind of bug in Dask. Though the exact nature has yet to be isolated.

**Comment by jakirkham** [comment]
Seeing some mention of `subgraph_callable` when debugging locally. May be related to PR ( https://github.com/dask/dask/pull/3998 ).

**Comment by jakirkham** [comment]
Reopened as this bug is likely an upstream issue with Dask and we should continue to track it until it can be moved to the Dask issue tracker and/or is fixed by a new version of Dask.

**Comment by jakirkham** [comment]
This was raised as issue ( https://github.com/dask/dask/issues/4178 ) and fixed by PR ( https://github.com/dask/dask/pull/4207 ). The fix was included in Dask  0.20.2+.

**Comment by jakirkham** [comment]
PR ( https://github.com/nanshe-org/nanshe_workflow/pull/297 ) reverted the workaround introduced in PR ( https://github.com/nanshe-org/nanshe_workflow/pull/272 ) and updated the Dask minimum dependency. As this is now resolved will close this issue.
